tests/Test.MSAL.NET.Unit/RequestsTests/ClientCredentialRequestTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Http;
+using Microsoft.Identity.Client.Internal.Instance;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.MSAL.NET.Unit.Mocks;
+
+namespace Test.MSAL.NET.Unit.RequestsTests
+{
+    [TestClass]
+    public class ClientCredentialRequestTests
+    {
+        private TokenCache _cache;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _cache = new TokenCache
+            {
+                ClientId = TestConstants.ClientId
+            };
+            Authority.ValidatedAuthorities.Clear();
+            HttpClientFactory.ReturnHttpClientForMocks = true;
+            HttpMessageHandlerFactory.ClearMockHandlers();
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            _cache.TokenCacheAccessor.AccessTokenCacheDictionary.Clear();
+            _cache.TokenCacheAccessor.RefreshTokenCacheDictionary.Clear();
+        }
+
+        [TestMethod]
+        [TestCategory("ClientCredentialRequestTests")]
+        public async Task ForceRefreshParameterTest()
+        {
+            TokenCacheHelper.PopulateCache(_cache.TokenCacheAccessor);
+
+            var parameters = new AuthenticationRequestParameters
+            {
+                Authority = Authority.CreateAuthority(TestConstants.AuthorityHomeTenant, false),
+                ClientId = TestConstants.ClientId,
+                Scope = TestConstants.Scope,
+                TokenCache = _cache,
+                RequestContext = new RequestContext(Guid.Empty),
+                User = new User
+                {
+                    Identifier = TestConstants.UserIdentifier,
+                    DisplayableId = TestConstants.DisplayableId
+                }
+            };
+
+            var clientCredentialRequest = new ClientCredentialRequest(parameters, false);
```

Comment:
you should put this test in ConfidentialClientTests. You should also be calling the top level API instead of Request class.